app/src/main/java/org/piwigo/ui/main/MainActivity.java

```diff
@@ -229,17 +245,36 @@ public void onRequestPermissionsResult(int requestCode,
         }
     }
 
-    private void selectPhoto(){
+    private void selectPhoto()
+    {
         Intent intent = new Intent();
         intent.setType("image/*");
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2)
             intent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true);
-        }
         intent.setAction(Intent.ACTION_GET_CONTENT);
         startActivityForResult(Intent.createChooser(intent,
                 getResources().getString(R.string.title_select_image)), SELECT_PICTURES);
     }
 
+    private void createAlbum(String catName)
+    {
+        int catId = 0;
+
+        Fragment f = getSupportFragmentManager().findFragmentById(R.id.content);
+        if (f instanceof AlbumsFragment) {
+            Integer cat = ((AlbumsFragment)f).getViewModel().getCategory();
+            if (cat != null)
+                catId = cat;
```

Comment:
What about the else? When does it happen that cat is null and how shall it be handled? Is it good to use catId = 0 in those cases?

Reply:
`catId = 0` means that there is no parent album, so I think this is pretty relevant to use this here.

Reply:
yes ok, but when should this happen? Maybe we always should have a Category in the viewmodel?